.pipelines/templates/.builder-release-template.yaml

```diff
@@ -111,58 +111,114 @@ steps:
-        -w /go/src/github.com/Azure/AgentBaker \
-        -e SUBSCRIPTION_ID="${SUBSCRIPTION_ID}" \
-        -e AZURE_RESOURCE_GROUP_NAME=${AZURE_RESOURCE_GROUP_NAME} \
-        -e BLOB_STORAGE_NAME=${BLOB_STORAGE_NAME} \
-        -e CLASSIC_BLOB=${CLASSIC_BLOB} \
-        -e BUILD_ID=$(Build.BuildId) \
-        -e MANAGED_SIG_ID=${MANAGED_SIG_ID} \
-        -e PACKER_BUILD_LOCATION=${PACKER_BUILD_LOCATION} \
-        -e OS_VERSION=${OS_VERSION} \
-        -e OS_SKU=${OS_SKU} \
-        -e OS_TYPE="Linux" \
-        -e VHD_DEBUG=${VHD_DEBUG} \
-        -e FEATURE_FLAGS=$(FEATURE_FLAGS) \
-        -e ARCHITECTURE=${ARCHITECTURE} \
-        -e ENABLE_TRUSTED_LAUNCH=${ENABLE_TRUSTED_LAUNCH} \
-        ${CONTAINER_IMAGE} make -f packer.mk scanning-vhd
-    retryCountOnTaskFailure: 2
-    displayName: Scanning VHD
-  - script: |
-        OS_DISK_URI="$(cat packer-output | grep "OSDiskUri:" | cut -d " " -f 2)" && \
-        MANAGED_SIG_ID="$(cat packer-output | grep "ManagedImageSharedImageGalleryId:" | cut -d " " -f 2)" && \
-        docker run --rm \
-        -v ${PWD}:/go/src/github.com/Azure/AgentBaker \
-        -w /go/src/github.com/Azure/AgentBaker \
-        -e SUBSCRIPTION_ID="${SUBSCRIPTION_ID}" \
-        -e AZURE_RESOURCE_GROUP_NAME=${AZURE_RESOURCE_GROUP_NAME} \
-        -e OS_DISK_URI=${OS_DISK_URI} \
-        -e MANAGED_SIG_ID=${MANAGED_SIG_ID} \
-        -e PACKER_BUILD_LOCATION=${PACKER_BUILD_LOCATION} \
-        -e CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-"docker"} \
-        -e OS_VERSION=${OS_VERSION} \
-        -e OS_SKU=${OS_SKU} \
-        -e OS_TYPE="Linux" \
-        -e IMG_SKU=${IMG_SKU} \
-        -e MODE=$(MODE) \
-        -e VHD_DEBUG=${VHD_DEBUG} \
-        -e FEATURE_FLAGS=$(FEATURE_FLAGS) \
-        -e SIG_GALLERY_NAME=${SIG_GALLERY_NAME} \
-        -e SIG_IMAGE_NAME=${SIG_IMAGE_NAME} \
-        -e SIG_IMAGE_VERSION=${SIG_IMAGE_VERSION} \
-        -e ENABLE_FIPS=${ENABLE_FIPS} \
-        -e ARCHITECTURE=${ARCHITECTURE} \
-        -e ENABLE_TRUSTED_LAUNCH=${ENABLE_TRUSTED_LAUNCH} \
-        -e SGX_INSTALL=${SGX_INSTALL} \
-        -e ENABLE_CGROUPV2=${ENABLE_CGROUPV2} \
-        -e GIT_BRANCH=$(Build.SourceBranch) \
-        ${CONTAINER_IMAGE} make -f packer.mk test-building-vhd
-    displayName: Run VHD Tests
+        PKR_RG_NAME="$(cat packer-output | grep "ResourceGroupName" | cut -d "'" -f 2 | head -1)" && \
+        captured_sig_version="$(cat vhdbuilder/packer/settings.json | grep "captured_sig_version" | awk -F':' '{print $2}' | awk -F'"' '{print $2}')" && \
+        IMPORTED_IMAGE_NAME="$(cat vhdbuilder/packer/settings.json | grep "imported_image_name" | awk -F':' '{print $2}' | awk -F'"' '{print $2}')" && \
+        SIG_IMAGE_NAME="$(cat vhdbuilder/packer/settings.json | grep "sig_image_name" | awk -F':' '{print $2}' | awk -F'"' '{print $2}')" && \
+        IMAGE_NAME="${SIG_IMAGE_NAME}-${captured_sig_version}" && \
+        parallel --joblog log.txt --results results/ --retries 1 ::: \
```

Comment:
this seems hacky... though I guess from reading around a little bit there isn't a clean way to have certain steps within a single job execute in parallel at the ADO level

Reply:
if we decide this is the best way to accomplish this, I'd at least recommend putting all this into its own `.sh` file probably somewhere under `vhdbuilder` and simply reference that within this yaml template

Reply:
Yeah agreed with putting it within a script. Actively developing this right now, testing the feasibility of gnu parallel for this. 

I was originally using make's implementation for parallelism which was pretty simple and clean, but @djsly suggested we retry only failed targets instead of all targets if one fails which I think makes sense, so I'm looking for ways to accomplish that.